### PR TITLE
CORDA-654 MockServices no longer has hard-coded identities

### PR DESCRIPTION
**Pull request by andr3ej**

### CODE REVIEW

File: node/src/main/kotlin/net/corda/node/services/identity/IdentityServiceInternal.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package net.corda.node.services.identity
```
Comment (mnesbit) [inline node/src/main/kotlin/net/corda/node/services/identity/IdentityServiceInternal.kt]: Can we put internal API interfaces in the services/api folder. That folder is met to contain the intra-node API plane.

File: core/src/test/kotlin/net/corda/core/contracts/DummyContractV2Tests.kt [hunk 2]
```diff
@@ -23,7 +24,9 @@ class DummyContractV2Tests {
 
     @Test
     fun `upgrade from v1`() {
-        val services = MockServices()
+        val services = rigorousMock<ServicesForResolution>().also {
+            doReturn(rigorousMock<CordappProvider>()).whenever(it).cordappProvider
```
Comment (andr3ej) [inline core/src/test/kotlin/net/corda/core/contracts/DummyContractV2Tests.kt]: looks like the cordappProvider isn't actually used

File: core/src/test/kotlin/net/corda/core/flows/CollectSignaturesFlowTests.kt [hunk 3]
```diff
@@ -129,7 +129,7 @@ class CollectSignaturesFlowTests {
     @Test
     fun `fails when not signed by initiator`() {
         val onePartyDummyContract = DummyContract.generateInitial(1337, notary, alice.ref(1))
-        val miniCorpServices = MockServices(listOf("net.corda.testing.contracts"), MINI_CORP.name, MINI_CORP_KEY)
+        val miniCorpServices = MockServices(listOf("net.corda.testing.contracts"), rigorousMock(), MINI_CORP.name, MINI_CORP_KEY)
```
Comment (andr3ej) [inline core/src/test/kotlin/net/corda/core/flows/CollectSignaturesFlowTests.kt]: the identityService isn't used at all, a lot of tests are like this

File: finance/src/test/kotlin/net/corda/finance/contracts/CommercialPaperTests.kt [hunk 4]
```diff
@@ -224,19 +223,16 @@ class CommercialPaperTestsGeneric {
...
-    private val issuerServices = MockServices(DUMMY_CASH_ISSUER_KEY)
-
+    private val notaryServices = MockServices(rigorousMock(), MEGA_CORP.name, DUMMY_NOTARY_KEY)
+    private val issuerServices = MockServices(listOf("net.corda.finance.contracts"), rigorousMock(), MEGA_CORP.name, DUMMY_CASH_ISSUER_KEY)
     private lateinit var moveTX: SignedTransaction
-
-    //    @Test
-    @Ignore
-    fun `issue move and then redeem`() = withTestSerialization {
+    @Test
```
Comment (andr3ej) [inline finance/src/test/kotlin/net/corda/finance/contracts/CommercialPaperTests.kt]: the test works, so i've unignored it so it doesn't rot any more

File: core/src/test/kotlin/net/corda/core/serialization/TransactionSerializationTests.kt [hunk 5]
```diff
@@ -95,7 +94,7 @@ class TransactionSerializationTests {
                     Command(TestCash.Commands.Move(), DUMMY_KEY_2.public))
 
             val ptx2 = notaryServices.signInitialTransaction(tx2)
-            val dummyServices = MockServices(DUMMY_KEY_2)
+            val dummyServices = MockServices(rigorousMock(), MEGA_CORP.name, DUMMY_KEY_2)
```
Comment (andr3ej) [inline core/src/test/kotlin/net/corda/core/serialization/TransactionSerializationTests.kt]: MEGA_CORP was the default name in MockServices

File: core/src/test/kotlin/net/corda/core/transactions/LedgerTransactionQueryTests.kt [hunk 6]
```diff
@@ -21,7 +22,10 @@ class LedgerTransactionQueryTests {
     @Rule
     @JvmField
     val testSerialization = SerializationEnvironmentRule()
-    private val services: MockServices = MockServices()
+    private val keyPair = generateKeyPair()
+    private val services = MockServices(rigorousMock<IdentityServiceInternal>().also {
+        doReturn(null).whenever(it).partyFromKey(keyPair.public)
```
Comment (andr3ej) [inline core/src/test/kotlin/net/corda/core/transactions/LedgerTransactionQueryTests.kt]: not a well-known identity in MockServices

File: core/src/test/kotlin/net/corda/core/transactions/LedgerTransactionQueryTests.kt [hunk 7]
```diff
@@ -21,7 +22,10 @@ class LedgerTransactionQueryTests {
     @Rule
     @JvmField
     val testSerialization = SerializationEnvironmentRule()
-    private val services: MockServices = MockServices()
+    private val keyPair = generateKeyPair()
```
Comment (andr3ej) [inline core/src/test/kotlin/net/corda/core/transactions/LedgerTransactionQueryTests.kt]: what the no-arg MockServices constructor does

File: finance/src/test/kotlin/net/corda/finance/contracts/asset/CashTests.kt [hunk 8]
```diff
@@ -67,9 +69,11 @@ class CashTests {
     @Before
     fun setUp() {
         LogHelper.setLevel(NodeVaultService::class)
-        megaCorpServices = MockServices(listOf("net.corda.finance.contracts.asset"), MEGA_CORP.name, MEGA_CORP_KEY)
-        miniCorpServices = MockServices(listOf("net.corda.finance.contracts.asset"), MINI_CORP.name, MINI_CORP_KEY)
-        val notaryServices = MockServices(listOf("net.corda.finance.contracts.asset"), DUMMY_NOTARY.name, DUMMY_NOTARY_KEY)
+        megaCorpServices = MockServices(listOf("net.corda.finance.contracts.asset"), rigorousMock(), MEGA_CORP.name, MEGA_CORP_KEY)
+        miniCorpServices = MockServices(listOf("net.corda.finance.contracts.asset"), rigorousMock<IdentityServiceInternal>().also {
+            doNothing().whenever(it).justVerifyAndRegisterIdentity(argThat { name == MINI_CORP.name })
```
Comment (andr3ej) [inline finance/src/test/kotlin/net/corda/finance/contracts/asset/CashTests.kt]: the test doesn't use the identityService for anything else, so it's valid for this to just not blow up

File: finance/src/test/kotlin/net/corda/finance/contracts/asset/CashTests.kt [hunk 9]
```diff
@@ -774,8 +778,9 @@ class CashTests {
     // Double spend.
     @Test
     fun chainCashDoubleSpendFailsWith() {
-        val mockService = MockServices(listOf("net.corda.finance.contracts.asset"), MEGA_CORP.name, MEGA_CORP_KEY)
-
+        val mockService = MockServices(listOf("net.corda.finance.contracts.asset"), rigorousMock<IdentityServiceInternal>().also {
+            doReturn(MEGA_CORP).whenever(it).partyFromKey(MEGA_CORP_PUBKEY)
```
Comment (andr3ej) [inline finance/src/test/kotlin/net/corda/finance/contracts/asset/CashTests.kt]: MEGA_CORP was a well-known identity in MockServices

File: finance/src/test/kotlin/net/corda/finance/contracts/asset/ObligationTests.kt [hunk 10]
```diff
@@ -51,9 +54,13 @@ class ObligationTests {
...
     private val outState = inState.copy(beneficiary = AnonymousParty(BOB_PUBKEY))
-    private val miniCorpServices = MockServices(listOf("net.corda.finance.contracts.asset"), MINI_CORP.name, MINI_CORP_KEY)
-    private val notaryServices = MockServices(DUMMY_NOTARY_KEY)
-    private val mockService = MockServices(listOf("net.corda.finance.contracts.asset"))
+    private val miniCorpServices = MockServices(listOf("net.corda.finance.contracts.asset"), rigorousMock(), MINI_CORP.name, MINI_CORP_KEY)
+    private val notaryServices = MockServices(rigorousMock(), MEGA_CORP.name, DUMMY_NOTARY_KEY)
+    private val mockService = MockServices(listOf("net.corda.finance.contracts.asset"), rigorousMock<IdentityServiceInternal>().also {
+        doReturn(null).whenever(it).partyFromKey(ALICE_PUBKEY)
+        doReturn(null).whenever(it).partyFromKey(BOB_PUBKEY)
+        doReturn(MEGA_CORP).whenever(it).partyFromKey(MEGA_CORP_PUBKEY)
```
Comment (andr3ej) [inline finance/src/test/kotlin/net/corda/finance/contracts/asset/ObligationTests.kt]: i'm not sure what happens here is actually significant from the pov of the tests, but to be safe this preserves the MockServices behaviour

File: node/src/main/kotlin/net/corda/node/services/identity/IdentityServiceInternal.kt (hunk 1 above)
Comment (andr3ej) [inline node/src/main/kotlin/net/corda/node/services/identity/IdentityServiceInternal.kt]: i've put this next to its impls, it can easily be moved if moving it becomes necessary

File: node/src/test/java/net/corda/node/services/vault/VaultQueryJavaTests.java [hunk 11]
```diff
@@ -449,30 +449,28 @@ public void aggregateFunctionsSumByIssuerAndCurrencyAndSortByAggregateSum() {
...
-                assertThat(results.getOtherResults().get(7)).isEqualTo(EncodingUtils.toBase58String(getBOC_PUBKEY()));
+                assertThat(results.getOtherResults().get(7)).isEqualTo(CryptoUtils.toStringShort(getBOC_PUBKEY()));
                 assertThat(results.getOtherResults().get(8)).isEqualTo("USD");
                 assertThat(results.getOtherResults().get(9)).isEqualTo(100L);
-                assertThat(results.getOtherResults().get(10)).isEqualTo(EncodingUtils.toBase58String(getDUMMY_CASH_ISSUER().getParty().getOwningKey()));
+                assertThat(results.getOtherResults().get(10)).isEqualTo(CryptoUtils.toStringShort(getDUMMY_CASH_ISSUER().getParty().getOwningKey()));
                 assertThat(results.getOtherResults().get(11)).isEqualTo("USD");
 
             } catch (NoSuchFieldException e) {
-                e.printStackTrace();
```
Comment (andr3ej) [inline node/src/test/java/net/corda/node/services/vault/VaultQueryJavaTests.java]: unrelated fix: issuerParty doesn't exist so the asserts were skipped and because of this line the test appeared green

File: testing/node-driver/src/main/kotlin/net/corda/testing/NodeTestUtils.kt [hunk 12]
```diff
@@ -31,7 +32,7 @@ import java.nio.file.Path
  */
 @JvmOverloads
 fun ledger(
-        services: ServiceHub = MockServices(),
+        services: ServiceHub = MockServices(makeTestIdentityService(listOf(MEGA_CORP_IDENTITY, MINI_CORP_IDENTITY, DUMMY_CASH_ISSUER_IDENTITY, DUMMY_NOTARY_IDENTITY)), MEGA_CORP.name),
```
Comment (andr3ej) [inline testing/node-driver/src/main/kotlin/net/corda/testing/NodeTestUtils.kt]: i'm not sure what's going to happen here as a lot of tests depend on this code, so no mocks in this pr

File: tools/loadtest/src/main/kotlin/net/corda/loadtest/tests/NotaryTest.kt [hunk 13]
```diff
@@ -21,7 +23,7 @@ data class NotariseCommand(val issueTx: SignedTransaction, val moveTx: SignedTra
 val dummyNotarisationTest = LoadTest<NotariseCommand, Unit>(
         "Notarising dummy transactions",
         generate = { _, _ ->
-            val issuerServices = MockServices(DUMMY_CASH_ISSUER_KEY)
+            val issuerServices = MockServices(makeTestIdentityService(listOf(MEGA_CORP_IDENTITY, MINI_CORP_IDENTITY, DUMMY_CASH_ISSUER_IDENTITY, DUMMY_NOTARY_IDENTITY)), MEGA_CORP.name, DUMMY_CASH_ISSUER_KEY)
```
Comment (andr3ej) [inline tools/loadtest/src/main/kotlin/net/corda/loadtest/tests/NotaryTest.kt]: i don't know how to run this test, so this is as refactored as it gets

File: node/src/main/kotlin/net/corda/node/services/identity/IdentityServiceInternal.kt (hunk 1 above)
Comment (andr3ej) [reply node/src/main/kotlin/net/corda/node/services/identity/IdentityServiceInternal.kt]: Done
